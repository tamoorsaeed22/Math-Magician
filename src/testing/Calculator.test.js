import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

it('renders calculator correctly', () => {
  const tree = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
