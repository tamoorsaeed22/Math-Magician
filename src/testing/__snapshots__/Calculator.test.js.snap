// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator correctly 1`] = `
Array [
  <b
    style={
      Object {
        "color": "darkslategray",
        "fontSize": "2em",
        "marginTop": "2em",
      }
    }
  >
    Let's do some math!
  </b>,
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      <div
        className="current-operand"
        data-testid="output"
      >
        0
      </div>
    </div>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;
